cassovary-benchmarks/src/main/scala/com/twitter/cassovary/ShortestPathBenchmark.scala

```diff
@@ -0,0 +1,25 @@
+package com.twitter.cassovary
+
+import com.twitter.cassovary.algorithms.shortestpath.SingleSourceShortestPath
+import com.twitter.cassovary.graph.{DirectedGraph, Node}
+import scala.util.Random
+
+class ShortestPathBenchmark(graph: DirectedGraph[Node]) extends OperationBenchmark {
+
+  val nodeList = graph.toList
+  val random = new Random(System.currentTimeMillis())
+  def randomNodeId() = random.nextInt(nodeList.length)
```

Comment:
This materializes the graph node list which could be expensive. Consider just keeping val nodes = graph.iterator

and inside def operation() {...
val node = graph.next()
...
}